Coding conventions

nick7695 opened the pull request:
The following Files has been updated according to Coding Conventions:
1)receiverCadmium.hpp(sdam12)
2)subnetCadmium.hpp(sdam12)
3)senderCadmium.hpp(sdam12)
4)message.hpp (nick7695)
5)main.cpp(nick7695)
6)message.cpp(nick7695)

Please review and provide feedback on the same.

sdam12 committed:
Coding Conventions

Changed Following Files: receivercadmium.hpp,subnetcadmium.hpp and Networkcadmium.hpp

Co-Authored-By: Sagnik Banerjee <nick7695@users.noreply.github.com>

sdam12 committed:
Coding Conventions

Files Changed according to Coding Conventions:
1) message.hpp(icnclude\data_structures)
2) main.cpp(src\)
3) message.cpp (src\)

Co-Authored-By: sdam12 <sdam12@users.noreply.github.com>

abhinavachanta commented on src/main.cpp:
```diff
@@ -26,188 +26,226 @@
-        static std::ostream& sink(){          
-            return out_data;
-        }
+    using hclock=chrono::high_resolution_clock;
```
One suggestion is to add spaces before and after = sign to enhance code readability.

sdam12 replied:
Suggestion taken into account. Thank You

abhinavachanta commented on include/senderCadmium.hpp:
```diff
@@ -45,26 +45,26 @@ using namespace std;
         public:
             //Parameters to be overwriten when instantiating the atomic model
-            TIME   preparationTime;
+            TIME   preparation_time;
```
This is a constant and should be upper case.

sdam12 replied:
Changes done. Constants are in Upper case now. Thank You.

sdam12 committed:
Changes Requested for receiver,subnet and sender Cadmium files

Changes Requested for receiver,subnet and sender Cadmium files.
Constant in Capital and added guards to the headers.

sdam12 committed:
Changes in main.cpp

Changes in main.cpp: One suggestion is to add spaces before and after = sign to enhance code readability.

Co-Authored-By: sdam12 <sdam12@users.noreply.github.com>

abhinavachanta commented on include/data_structures/message.hpp:
```diff
@@ -10,19 +10,18 @@ using namespace std;
-        Message_t(); 
-    }
-};
+    struct Message_t{
```
Structure name should not start with a capital letter. So 'Message_t' should be changed to 'message_t'

nick7695 replied:
Its is a good suggestion, that structure name should not start by capital letter. But, no changes has been requested by professor in structure name. 
Even if I tried to do changes in structure name its breaking my code. I think it will be good if I we leave it without changing. Still please let me know what are your thoughts on that?

nick7695 replied:
Changes to the structure name has been done. Structure name changed from 'Message_t' to 'message_t'.

abhinavachanta reviewed (changes requested):
Structure name should not start with a capital letter. So 'Message_t' should be changed to 'message_t'

jrajeev459 commented on include/data_structures/message.hpp:
```diff
@@ -10,19 +10,18 @@ using namespace std;
+    ostream& operator<<(ostream& os, const Message_t& msg);
 
 
 #endif // BOOST_SIMULATION_MESSAGE_HPP
```
Need to have underscore in front of Boost and at the end of HPP

sdam12 committed:
Changes in Structure name from Message_t to message_t

Changes requested for Structure name. Name should be change from Message_t to message_t. Since Structure name cannot start with capital letter.

Co-Authored-By: sdam12 <sdam12@users.noreply.github.com>

sdam12 committed:
Structure name has been changed for receiver_defs,sender_defs and subnet_defs

Structure name has been changed for receiver_defs,sender_defs and subnet_defs.

Co-Authored-By: Sagnik Banerjee <nick7695@users.noreply.github.com>

abhinavachanta commented on test/src/receiver/main.cpp:
```diff
@@ -28,20 +28,20 @@ using TIME = NDTime;
+  ApplicationGen(const char* file_path) : iestream_input<message_t,T>(file_path) {}
 };
 
 
```
Many lines in this file are more than 80 characters in length (example: line 60-67)

sdam12 replied:
Changes done. Thank You.

abhinavachanta commented on test/src/sender/main.cpp:
```diff
@@ -105,13 +105,13 @@ cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp_data),typeid(outp_pa
 cadmium::dynamic::modeling::ICs ics_TOP = {
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Sender_defs::controlIn>("generator_con","sender1"),
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Sender_defs::ackIn>("generator_ack","sender1")
+  cadmium::dynamic::translate::make_IC<iestream_input_defs<message_t>::out,sender_defs::controlIn>("generator_con","sender1"),
```
Many lines in this file are more than 80 characters in length (like this line)

sdam12 replied:
Changes done. Thank You.

abhinavachanta commented on test/src/subnet/main.cpp:
```diff
@@ -96,10 +96,10 @@ cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp_out)};
 };
 cadmium::dynamic::modeling::ICs ics_TOP = {
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Subnet_defs::in>("generator","subnet1")
+  cadmium::dynamic::translate::make_IC<iestream_input_defs<message_t>::out,subnet_defs::in>("generator","subnet1")
```
Many lines in this file are more than 80 characters in length (like this line)

sdam12 replied:
Changes done. Thank You.

abhinavachanta commented:
1)receiverCadmium.hpp
2)subnetCadmium.hpp
3)senderCadmium.hpp

As prof suggested, pls use lowercase letters for files names

abhinavachanta reviewed (changes requested):
Pls check my comments and make required changes

jrajeev459 commented on include/senderCadmium.hpp:
```diff
@@ -73,102 +73,112 @@ using namespace std;
+                        state.model_active = true; 
+                        state.next_internal = PREPARATION_TIME;   
+                    } 
+				    else {
```
align  else{  here and in line 97,112 , 136 any others

jrajeev459 commented on include/senderCadmium.hpp:
```diff
@@ -73,102 +73,112 @@ using namespace std;
+                    }
+                }   
+                return bags;
+		    }
```
parenthesis } should be align properly

sdam12 committed:
Changes in number of characters per line and algined as per user readability

Changes in number of characters per line and algined as per user readability. So that lay user to code can read code easily.

Co-Authored-By: Sagnik Banerjee <nick7695@users.noreply.github.com>

jrajeev459 commented on include/sender_cadmium.hpp:
```diff
@@ -0,0 +1,187 @@
+
+	//Port definition
+    struct sender_defs{
+        struct packetSentOut : public out_port<message_t> {
```
Add underscore to struct  packet_sent_out  and do the same for remaining

abhinavachanta commented on test/src/receiver/main.cpp:
```diff
@@ -96,10 +96,10 @@ cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp)};
 };
 cadmium::dynamic::modeling::ICs ics_TOP = {
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Receiver_defs::in>("generator","receiver1")
+  cadmium::dynamic::translate::make_IC<iestream_input_defs<message_t>::out,receiver_defs::in>("generator","receiver1")
```
line length > 80. 
Please break into smaller lines.
Many of the lines in your test files have length>80. Please check and correct.

nick7695 replied:
Changes to the main.cpp has been done. Lines updated with this commit is mentioned in the commit description.
Thank You.

abhinavachanta reviewed (changes requested):
Many of the lines in your test cpp files have length>80. Please check and correct.

sdam12 committed:
Changed character per line less than 80.

Number of character per line is less than 80.
Lines changed:
149-152.
178-182
187-190
193-200
218-221

Co-Authored-By: sdam12 <sdam12@users.noreply.github.com>

jrajeev459 reviewed (approved):
All file are okay now, do the comments done by abhinav.
